### PR TITLE
Initial extension field (de)serialization logic.

### PR DESCRIPTION
**Pull request by davidv1992**

### CODE REVIEW

File: ntp-proto/src/packet.rs [hunk 1]
```diff
@@ -275,11 +441,16 @@ impl NtpPacket {
...
+                } else {
+                    None
+                };
                 Ok(NtpPacket {
                     header: NtpHeader::V4(header),
+                    efdata,
+                    mac,
                 })
             }
             _ => Err(PacketParsingError::InvalidVersion(version)),
```
Comment (folkertdev) [inline ntp-proto/src/packet.rs:456]: we also want to support version 3 now right?

Comment (davidv1992) [reply ntp-proto/src/packet.rs:456]: I will deal with that when rebasing, but in short, version 3 does not support extension fields, only a des-based mac.

File: ntp-proto/src/packet.rs [hunk 2]
```diff
@@ -108,8 +107,163 @@ impl NtpAssociationMode {
...
+                    field.serialize(w)?;
+                }
+                Ok(())
+            }
+        }
+    }
+
+    fn deserialize(data: &'a [u8]) -> Result<(ExtensionFieldData<'a>, usize), PacketParsingError> {
+        let mut offset = 0;
+        while data.len() - offset >= 28 {
```
[the remark this answers is not in the document]
Comment (davidv1992) [reply ntp-proto/src/packet.rs:226]: Made it a more descriptive constant.

File: ntp-proto/src/packet.rs [hunk 3]
```diff
@@ -264,8 +418,20 @@ impl NtpHeaderV4 {
...
+impl<'a> NtpPacket<'a> {
+    pub fn into_owned(self) -> NtpPacket<'static> {
+        NtpPacket::<'static> {
+            header: self.header,
+            efdata: self.efdata.into_owned(),
+            mac: self.mac.map(|v| v.into_owned()),
+        }
+    }
+
+    pub fn ef_iter<'b: 'a>(&'b self) -> impl Iterator<Item = Cow<'b, ExtensionField<'a>>> + 'b {
```
[the remark this answers is not in the document]
Comment (davidv1992) [reply ntp-proto/src/packet.rs:430]: done

**Review by folkertdev — APPROVED** [review]

**Pull request merged**